### ISSUE TITLE
Default custom editor isn't used if opening a file while extensions are activating

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#117145]
Repro:

1. Open VS Code where the active editor is using a custom editor
2. Open the same file type, **you will be presented with the unsupported binary format page**
3. Open the same file type after extensions have finished activating and it will work

![recording (4)](https://user-images.githubusercontent.com/2193314/108608270-c730ca80-737a-11eb-84fb-20495d41eac3.gif)

**Comment by mjbvz** [comment]
I just added a 10sec delay to the activation of the builtin image preview extension (with an await) but could not repo this

Does Luna paint make sure that `registerCustomEditorProvider` is called before `activate` returns?

**Comment by mjbvz** [comment]
Nvm, looks like we fire the activation event but it returns right away for the second call, even if the first has not completed yet

**Comment by mjbvz** [comment]
To test, you can use the image editor:

1.  Apply the diff to artificially delay to image editor resolving by 10 seconds

```
diff --git a/extensions/image-preview/src/extension.ts b/extensions/image-preview/src/extension.ts
index 10722360dd5..4ae5f23c238 100644
--- a/extensions/image-preview/src/extension.ts
+++ b/extensions/image-preview/src/extension.ts
@@ -9,7 +9,9 @@ import { PreviewManager } from './preview';
 import { SizeStatusBarEntry } from './sizeStatusBarEntry';
 import { ZoomStatusBarEntry } from './zoomStatusBarEntry';
 
-export function activate(context: vscode.ExtensionContext) {
+export async function activate(context: vscode.ExtensionContext) {
+
+	await new Promise(resolve => setTimeout(resolve, 10000));
 	const sizeStatusBarEntry = new SizeStatusBarEntry();
 	context.subscriptions.push(sizeStatusBarEntry);
 ```

2. Follow Daniel's steps

**Comment by bpasero** [comment]
I am not sure why this listener is only added after restore:

https://github.com/bpasero/vscode/blob/763e089ae54efdc3e6382699e7391c6df8b6f2be/src/vs/workbench/services/editor/browser/editorService.ts#L166-L168

We can try to change that but we have to be careful not causing deadlocks because it is possible that something is waiting on the editor restore that is needed to restore custom editors from extensions.

**Comment by mjbvz** [comment]
Actually looks like we already have an issue tracking this for notebooks.

Duplicate of #99369
